tables_design.md

```diff
@@ -0,0 +1,152 @@
+
+As a shop manager
+So I can manage items
+I want to be able to create a new item.
+
+As a shop manager
+So I can know which orders were made
+I want to keep a list of orders with their customer name.
+
+As a shop manager
+So I can know which orders were made
+I want to assign each order to their corresponding item.
+
+As a shop manager
+So I can know which orders were made
+I want to know on which date an order was placed. 
+
+As a shop manager
+So I can manage orders
+I want to be able to create a new order.
+```
+
+```
+Nouns:
+
+list of shop items, name, unit price, quantity
+list of orders, customer name, order item, order date
+
+Methods:
+List all shop items
+Create a new shop item
+List of all orders
+Create a new order
+
+```
+
+## 2. Infer the Table Name and Columns
+
+Put the different nouns in this table. Replace the example with your own nouns.
+
+| Record                | Properties          |
+| --------------------- | ------------------  |
+| items                 | name, unit_price, quantity, order_id
+| orders                | customer_name, date
+
+1. Name of the first table (always plural): `items` 
+
+    Column names: `name`, `unit_price`, `quantity`
+
+2. Name of the second table (always plural): `orders` 
+
+    Column names: `customer_name`, `date`, `item_id`
+
+## 3. Decide the column types.
+
+[Here's a full documentation of PostgreSQL data types](https://www.postgresql.org/docs/current/datatype.html).
+
+Most of the time, you'll need either `text`, `int`, `bigint`, `numeric`, or `boolean`. If you're in doubt, do some research or ask your peers.
+
+Remember to **always** have the primary key `id` as a first column. Its type will always be `SERIAL`.
+
+```
+# EXAMPLE:
+
+Table: items
+id: SERIAL
+name: text
+unit_price: numeric
+quanitity: int
+order_id: int
+
+Table: orders
+id: SERIAL
+customer_name: text
+date: date
+```
+
+## 4. Decide on The Tables Relationship
+
+Most of the time, you'll be using a **one-to-many** relationship, and will need a **foreign key** on one of the two tables.
+
+To decide on which one, answer these two questions:
+
+1. Can one [TABLE ONE] have many [TABLE TWO]? (Yes/No)
+2. Can one [TABLE TWO] have many [TABLE ONE]? (Yes/No)
+
+You'll then be able to say that:
+
+1. **[A] has many [B]**
+2. And on the other side, **[B] belongs to [A]**
+3. In that case, the foreign key is in the table [B]
+
+Replace the relevant bits in this example with your own:
+
+```
+# EXAMPLE
+
+1. Can one item have many orders? NO
+2. Can one order have many items? YES
+
+-> Therefore,
+-> An order HAS MANY items
+-> An item BELONGS TO an order
+
+-> Therefore, the foreign key is on the items table.
+```
+
+*If you can answer YES to the two questions, you'll probably have to implement a Many-to-Many relationship, which is more complex and needs a third table (called a join table).*
+
+## 4. Write the SQL.
+
+```sql
+CREATE TABLE orders (
+  id SERIAL PRIMARY KEY,
+  customer_name text,
+  date date
```

Comment:
'date' could have an ever-so-slightly more specific name, like 'order_date' just to differentiate more easily in instances like this one :)